source/common/router/router.cc

```diff
@@ -698,6 +705,46 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
       *this, std::move(generic_conn_pool), can_send_early_data, can_use_http3);
   LinkedList::moveIntoList(std::move(upstream_request), upstream_requests_);
   upstream_requests_.front()->acceptHeadersFromRouter(end_stream);
+  if (streaming_shadows_) {
+    // start the shadow streams.
+    for (const auto& shadow_policy_wrapper : active_shadow_policies_) {
+      const auto& shadow_policy = shadow_policy_wrapper.get();
+      const absl::optional<absl::string_view> shadow_cluster_name =
+          getShadowCluster(shadow_policy, *downstream_headers_);
+      if (!shadow_cluster_name.has_value()) {
+        continue;
+      }
+      auto shadow_headers = Http::createHeaderMap<Http::RequestHeaderMapImpl>(*shadow_headers_);
+      auto options = Http::AsyncClient::RequestOptions()
+                         .setTimeout(timeout_.global_timeout_)
+                         .setParentSpan(callbacks_->activeSpan())
+                         .setChildSpanName("mirror")
+                         .setSampled(shadow_policy.traceSampled())
+                         .setIsShadow(true)
+                         .setBufferAccount(callbacks_->account())
+                         .setBufferLimit(callbacks_->decoderBufferLimit());
```

Comment:
@alyssawilk @snowp what do you think the default limit should be here? I was thinking either `callbacks_->decoderBufferLimit()` or we could be more aligned to previous behavior and do `max(retry_shadow_buffer_limit_, 1)`.

Reply:
Neither are perfect choices IMO, maybe retry_shadow_buffer_limit so that users can still impact how much memory is used for shadowing without changing the whole decoder buffer?